app/src/main/java/net/gsantner/markor/ui/hleditor/ListHandler.java

```diff
@@ -0,0 +1,54 @@
+package net.gsantner.markor.ui.hleditor;
+
+import android.text.Editable;
+import android.text.Spannable;
+import android.text.Spanned;
+import android.text.TextWatcher;
+
+import net.gsantner.markor.format.markdown.MarkdownHighlighterPattern;
+import net.gsantner.markor.util.StringUtils;
+
+import java.util.regex.Matcher;
+
+public class ListHandler implements TextWatcher {
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+        if ( count > 0 && start > -1 && start < s.length() && s.charAt(start) == '\n' ) {
```

Comment:
Detects if enter pressed on empty list (correctly handles indent) and marks line for deletion.

Reply:
You can add exactly this comment as comment in code above the line